browser/model/jbosseap.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+let path = require('path');
+let fs = require('fs-extra');
+var rimraf = require('rimraf');
+
+import JbosseapAutoInstallGenerator from './jbosseap-autoinstall';
+import InstallableItem from './installable-item';
+import Installer from './helpers/installer';
+import Logger from '../services/logger';
+import JdkInstall from './jdk-install';
+import Platform from '../services/platform';
+
+class JbosseapInstall extends InstallableItem {
+  constructor(installerDataSvc, targetFolderName, downloadUrl, fileName, jbosseapSha256) {
+    super(JbosseapInstall.KEY, downloadUrl, fileName, targetFolderName, installerDataSvc, true);
+
+    this.sha256 = jbosseapSha256;
+    this.installConfigFile = path.join(this.installerDataSvc.tempDir(), 'jbosseap-autoinstall.xml');
+    this.ConfigFile = path.join(this.installerDataSvc.tempDir(), 'jbosseap-autoinstall.xml.variables');
+    this.runtimeproperties = path.join(this.installerDataSvc.installRoot, '\\developer-studio\\studio', 'runtime_locations.properties');
```

Comment:
Unless you call the setup method, all subfolders in installerDataSvc are undefined. But that's called when you click Download & Install.